src/request/order.ts

```diff
@@ -0,0 +1,40 @@
+import isURL = require('validator/lib/isURL');
+import { ArgumentError } from '../errors';
+
+interface OrderProps {
+  amount?: number;
+  currency?: string;
+  discountCode?: string;
+  affiliateId?: string;
+  subaffiliateId?: string;
+  referrerUri?: string;
```

Comment:
Technically, this can be a `URL` type, provided you add `import { URL } from 'url';`. This will have some cascading changes, with an issue on line 32 though.